alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ProviderConfigSelectCustomFunctionAction.java

```diff
@@ -66,14 +66,21 @@ public ProviderConfigSelectCustomFunctionAction(AuthorizationManager authorizati
         if (descriptorKey.isPresent()) {
             List<ConfigurationModel> configurationModels = configurationAccessor.getConfigurationsByDescriptorKeyAndContext(descriptorKey.get(), ConfigContextEnum.GLOBAL);
             options = configurationModels.stream()
-                          .map(ConfigurationModel::getCopyOfKeyToFieldMap)
-                          .map(FieldAccessor::new)
-                          .map(accessor -> accessor.getString(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME))
+                          .map(this::createNameToIdOption)
                           .flatMap(Optional::stream)
-                          .map(LabelValueSelectOption::new)
                           .collect(Collectors.toList());
         }
         LabelValueSelectOptions optionList = new LabelValueSelectOptions(options);
         return new ActionResponse<>(HttpStatus.OK, optionList);
     }
+
+    private Optional<LabelValueSelectOption> createNameToIdOption(ConfigurationModel configurationModel) {
+        Optional<String> providerConfigName = configurationModel.getField(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME).flatMap(ConfigurationFieldModel::getFieldValue);
+        if (providerConfigName.isPresent()) {
+            LabelValueSelectOption providerConfigOption = new LabelValueSelectOption(providerConfigName.get(), configurationModel.getConfigurationId().toString());
+            return Optional.of(providerConfigOption);
+        }
+        return Optional.empty();
```

Comment:
Could the optional mapping be used here instead of these if checks? We start with an optional and end with one. I feel there's a cleaner way to do this.

Reply:
Couldn't we use the `.map()` method on the providerConfigName variable to convert it to a LabelValueSelectOption and then you won't need the if's